### PR TITLE
Vector_container tests

### PR DESCRIPTION
**Pull request by YakhtinLeonid**

### CODE REVIEW

**Review by orrrrtem — APPROVED** [review]

File: cpp/oneapi/dal/graph/test/container_test.cpp [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+    {
+        CountingAllocator<T> alloc;
+        vector_container<T, CountingAllocator<T>> vec(n, alloc);
+        REQUIRE(!vec.empty());
+        REQUIRE(vec.get_count() == n);
+        REQUIRE(vec.size() == n);
+        REQUIRE(vec.capacity() == lower_bound_pow(n));
+    }
+    // TODO: Uncomment after the memory leak fix
+    // REQUIRE(allocated_bytes_count == 0);
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/graph/test/container_test.cpp:157]: It seems strange to me that memory leak can affect number of allocated bytes in this test.

**Pull request merged**